vernac/vernacentries.ml

```diff
@@ -634,7 +634,11 @@ let find_source (u,k,v as cst) src =
   else Option.default [] (search src cst)
 
 let pr_constraint_source = function
-  | GlobRef ref -> pr_global ref
+  | GlobRef ref -> begin try pr_global ref
+      with Not_found ->
+        (* global in a module type or functor *)
+        GlobRef.print ref
```

Comment:
Maybe we should have a distinguishing notation here. This is for debug, and sometimes we may confuse a constant bound in a functor with a global one.

Reply:
Bound constants are never constraint sources